src/main/java/org/ibp/api/java/impl/middleware/common/validator/GermplasmUpdateValidator.java

```diff
@@ -0,0 +1,171 @@
+package org.ibp.api.java.impl.middleware.common.validator;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.generationcp.commons.util.DateUtil;
+import org.generationcp.middleware.api.attribute.AttributeDTO;
+import org.generationcp.middleware.api.breedingmethod.BreedingMethodService;
+import org.generationcp.middleware.api.germplasm.GermplasmNameTypeDTO;
+import org.generationcp.middleware.domain.germplasm.GermplasmUpdateDTO;
+import org.generationcp.middleware.manager.api.GermplasmDataManager;
+import org.generationcp.middleware.manager.api.LocationDataManager;
+import org.generationcp.middleware.pojos.Germplasm;
+import org.generationcp.middleware.pojos.Location;
+import org.ibp.api.java.germplasm.GermplasmService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.BindingResult;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Component
+public class GermplasmUpdateValidator {
+
+	@Autowired
+	private GermplasmService germplasmService;
+
+	@Autowired
+	private org.generationcp.middleware.api.germplasm.GermplasmService germplasmMiddlewareService;
+
+	@Autowired
+	private GermplasmDataManager germplasmDataManager;
+
+	@Autowired
+	private LocationDataManager locationDataManager;
+
+	@Autowired
+	private BreedingMethodService breedingMethodService;
+
+	public void validateEmptyList(final BindingResult errors, final List<GermplasmUpdateDTO> germplasmUpdateDTOList) {
+		if (germplasmUpdateDTOList == null || germplasmUpdateDTOList.isEmpty()) {
+			errors.reject("germplasm.update.empty.list", "");
+		}
+	}
+
+	public void validateCodes(final BindingResult errors, final List<GermplasmUpdateDTO> germplasmUpdateDTOList) {
```

Comment:
Maybe I am missing something with this function. How an user get ambiguosCodes? 
You will have a file and how given the columns you will build the names and attributes map?

Reply:
This is specifically for the case where an FCODE is defined as Name and Attribute in the udflds. Yeah I think we can move this checking in the client when building the map for names and attributes.